docs/src/explanations/memories.md

```diff
@@ -114,6 +114,28 @@ Here is an example single read/write port waveform, with [masks](#masks) (again,
 
 ![read/write ports example waveform](https://svg.wavedrom.com/github/freechipsproject/www.chisel-lang.org/master/docs/src/main/resources/json/smem_rw.json)
 
+Single-ported SRAMs can also be explicitly generated by using the `readWrite` call, which yields a single read/write accessor like so:
+
+```scala mdoc:silent
+class RDWR_Smem extends Module {
+  val width: Int = 32
+  val io = IO(new Bundle {
+    val enable = Input(Bool())
+    val write = Input(Bool())
+    val addr = Input(UInt(10.W))
+    val dataIn = Input(UInt(width.W))
+    val dataOut = Output(UInt(width.W))
+  })
+
+  val mem = SyncReadMem(1024, UInt(width.W))
+  val rwPort = mem.readWrite(io.addr, io.dataIn, io.enable, io.write)
+
+  when(io.enable && !io.write) {
+    io.dataOut := rwPort
```

Comment:
this seems wrong. Isn't it off by a cycle? I think we should just assign io.dataOut := rwPort without any when